agent/src/beerocks/slave/tasks/topology_task.cpp

```diff
@@ -207,14 +207,152 @@ void TopologyTask::handle_topology_query(ieee1905_1::CmduMessageRx &cmdu_rx,
         return;
     }
 
+    if (!add_device_information_tlv()) {
```

Comment:
you should have split the "move to functions" and the changes in the implementation to 2 separate commits - this is really hard to review

Reply:
But then it would have break it...